graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/nodes/cast/JSToUInt32Node.java

```diff
@@ -179,6 +181,16 @@ protected int doBigInt(@SuppressWarnings("unused") BigInt value) {
         throw Errors.createTypeErrorCannotConvertBigIntToNumber(this);
     }
 
+    @Specialization
+    protected final Number doRecord(@SuppressWarnings("unused") Record value) {
+        throw Errors.createTypeErrorCannotConvertToNumber("a Record value", this);
```

Comment:
have seen this a few times already. Maybe it is worth extracting this to a separate `Errors.createTypeErrorCannotConvertRecordToNumber(this)` and same for Tuple?

Reply:
This call isn't unique to Record and Tuple, also Symbol and BigInt follows this pattern.
There are 8 places we do the same checks (was very confusing to me)
- JSToUInt32Node
- JSToInt32Node
- JSToNumberNode
- JSToDoubleNode
- JSToIntegerAsIntNode
- JSToIntegerAsLongNode
- JSToIntegerOrInfintiyNode
- JSRuntime

I'm wondering if we can derive them from `JSToNumberNode` OR add a common `JSAbstractToNumberNode` base where those cases are implemented just once... 
(I didn't change it yet as I payed attention to not changing to much existing code and keeping the MR small)

Reply:
`JSToNumberNode` is a sibling of the others (there is a `ToNumber` abstract operation in JavaScript) so I wouldn't make them derive from each other. Having a common base class could make sense, but agree, let's not do that in this PR.

Extracting the method (and avoiding to pass in the same string as argument several times) would just improve readability a bit, but no big thing in this case.